### ISSUE TITLE
Loopback 3.x - Breaking create

### ISSUE THREAD

**Issue by elkorep** [strongloop/loopback#2642]
With a loopback 3.x app, if a model does not contain an ID property, and uses autogenerated ID, the create operation fails.

Steps to Reproduce:
1. Create a loopback app using 3.x and the 'notes' template
2. On the Note model, attempt a POST crud providing an ID. This will return an error response.

**Comment by Amir-61** [comment]
Connect to https://github.com/strongloop/loopback/issues/2644

**Comment by Amir-61** [comment]
@elkorep Could you please confirm what error message you were seeing?
Was it this:
```
loopback deprecated loopback#context middleware is deprecated....
```

**Comment by bajtos** [comment]
This is expected, see https://github.com/strongloop/loopback-datasource-juggler/pull/982.

If your Model is using an auto-generated id, then we don't allow the clients to provide id values. This is needed to prevent a DoS attack, where the client can provide a MAX_INT_VALUE in id, and thus triggering integer overflow in subsequent create calls. See https://github.com/strongloop/loopback-datasource-juggler/issues/849

@jannyHou I am assigning this issue to you. I think we need to add an entry to 3.0-RELEASE-NOTES.md in juggler and/or update related documentation.

cc @crandmck

**Comment by bajtos** [comment]
@elkorep If you want to allow clients to provide a custom id value, then you need to set `forceId:false` in your model JSON file.

```js
{
  "name": "Notes",
  "forceId": false,
  // ...
}
```

**Comment by jannyHou** [comment]
Closing since pr https://github.com/strongloop/loopback-datasource-juggler/pull/1103 merged.
And it's also included in our new doc [Migrating-to-3.0](http://loopback.io/doc/en/lb3/Migrating-to-3.0.html)
